src/main/java/de/tum/cit/aet/artemis/atlas/service/competency/CompetencyProgressService.java

```diff
@@ -308,12 +292,15 @@ private void calculateConfidence(Set<CompetencyExerciseMasteryCalculationDTO> ex
         double recencyConfidenceHeuristic = calculateRecencyConfidenceHeuristic(participantScoreInfos);
         double difficultyConfidenceHeuristic = calculateDifficultyConfidenceHeuristic(participantScoreInfos, exerciseInfos);
         double quickSolveConfidenceHeuristic = calculateQuickSolveConfidenceHeuristic(participantScoreInfos);
+        double competencyLinkWeightConfidenceHeuristic = calculateCompetencyLinkWeightConfidenceHeuristic(participantScoreInfos);
 
         // Standard factor of 1 (no change to mastery compared to progress) plus the confidence heuristics
-        double confidence = DEFAULT_CONFIDENCE + recencyConfidenceHeuristic + difficultyConfidenceHeuristic + quickSolveConfidenceHeuristic;
+        double confidence = DEFAULT_CONFIDENCE + recencyConfidenceHeuristic + difficultyConfidenceHeuristic + quickSolveConfidenceHeuristic
+                + competencyLinkWeightConfidenceHeuristic;
+        confidence = Math.clamp(confidence, 0, 2);
```

Comment:
Why do we clamp it?
I think it makes sense to add a variable with an explanation of the upper bound of 2 with the reasoning why this value was chosen